src/parser/instruction.rs

```diff
@@ -399,7 +436,20 @@ mod tests {
                     name: "custom".to_owned(),
                     parameters: HashMap::new()
                 })
-            }
+            },
+            Instruction::Pulse {
+                blocking: true,
+                frame: FrameIdentifier {
+                    name: "xy".to_owned(),
+                    qubits: vec![Qubit::Fixed(0)]
+                },
+                waveform: Box::new(WaveformInvocation {
```

Comment:
Does the `WaveformInvocation` take up enough space that it should be heap-allocated, ergo `Box::new`?

Reply:
Ah, there must have been a self-referencing type here at some point in the past, but I don't see a need for it now. I'll make this its own issue.

Reply:
Issue: https://github.com/rigetti/quil-rust/issues/15

Reply:
Cheers, I'll put together a happy little PR.